qualtran/bloqs/basic_gates/rotation.py

```diff
@@ -319,6 +466,83 @@ def __str__(self):
         return f'Rz({self.angle})'
 
 
+@bloq_example
+def _rz() -> Rz:
+    a = sympy.Symbol('a')
+    rz = Rz(a)
+    return rz
+
+
+_RZ_DOC = BloqDocSpec(bloq_cls=Rz, examples=[_rz], call_graph_example=None)
+
+
+@frozen
+class CRz(Bloq):
```

Comment:
please override the ctrl system for this (and other controlled bloqs), and add a test to verify gate counts for `CRz().controlled()` (should be `CRz() + And() + And.adjoint()`)

Reply:
why would I need to override the ctrl system? there isn't a native CCRz bloq

Reply:
Because right now, `CRz.controlled` would return `Controlled(CRz)`, i.e. does not merge the two controls using `And`. 

The second case of `get_ctrl_system_1bit_cv` essentially handles this

Reply:
(sync discussion) I did not notice the decomposition, in which case the override is not needed.

But I still think it's good to override to get a much more cost effective circuit. The current `CRz(2*pi*t).controlled()` gives
![download](https://github.com/user-attachments/assets/bac6c838-b057-4d37-ab98-db9f241cd3a0)
`toffoli: 2, and_bloq: 2, clifford: 2, rotation: 2, measurement: 2`

But with the override it becomes
![download](https://github.com/user-attachments/assets/2f221cf8-886b-46e8-b46a-2cef18c37485)
`and_bloq: 1, clifford: 3, rotation: 2, measurement: 1`

---
full code

```py
import attrs
import numpy as np
import sympy

from qualtran import Bloq, DidNotFlattenAnythingError
from qualtran.bloqs.basic_gates import CRz
from qualtran.bloqs.mcmt import And
from qualtran.drawing import show_bloq
from qualtran.resource_counting import get_cost_value, QECGatesCost

def display_bloq_and_cost(bloq, *, flat=True):
    def _pred(binst):
        return not binst.bloq_is(And)
        
    cbloq = bloq.as_composite_bloq()
    if flat: cbloq = cbloq.flatten(pred=_pred)

    show_bloq(cbloq, 'latex')
    cost = get_cost_value(bloq, QECGatesCost())
    print(cost)

theta = 2 * np.pi * sympy.Symbol("t")
crz = CRz(theta)
display_bloq_and_cost(crz.controlled())

@attrs.frozen
class CRzWithOverride(CRz):
    def get_ctrl_system(self, ctrl_spec):
        from qualtran.bloqs.mcmt.specialized_ctrl import get_ctrl_system_1bit_cv_from_bloqs

        return get_ctrl_system_1bit_cv_from_bloqs(
            self,
            ctrl_spec,
            current_ctrl_bit=1,
            bloq_with_ctrl=self,
            ctrl_reg_name='ctrl'
        )

display_bloq_and_cost(CRzWithOverride(theta).controlled())
print(cost.total_t_count())
```

Reply:
that can be done in a follow-up. I don't think it's worth blocking this PR any more than it already has been